doc/source/cross-language.rst

```diff
@@ -0,0 +1,246 @@
+
+    public static void main(String[] args) {
+      Ray.init();
+      
+      // Define a Python class.
+      PyActorClass actorClass = PyActorClass.of(
+          "<your python module>", "Counter");
+      
+      // Create a Python actor and call actor method.
+      PyActorHandle actor = Ray.actor(actorClass).remote();
+      ObjectRef<Integer> objRef1 = actor.task(
+          PyActorMethod.of("increment", Integer.class)).remote();
+      Assert.assertEquals(objRef1.get(), 1);
+      ObjectRef<Integer> objRef2 = actor.task(
+          PyActorMethod.of("increment", Integer.class)).remote();
+      Assert.assertEquals(objRef2.get(), 2);
+
+      // Define a Python remote function.
+      PyFunction<Integer> addFunction = PyFunction.of(
+          "<your python module>", "add", Integer.class);
+
+      // Call the Python remote function.
+      ObjectRef<Integer> objRef3 = Ray.task(add_function, 1, 2).remote();
+      Assert.assertEquals(objRef3.get(), 3);
+
+      Ray.shutdown();
+    }
+  }
+
+Cross-language data serialization
+---------------------------------
+
+The arguments and return values of ray call can be serialized & deserialized
+automatically if their types are the following:
+  
+  - Primitive data types
+      ===========   =======  =======
+      MessagePack   Python   Java
+      ===========   =======  =======
+      nil           None     null
+      bool          bool     Boolean
+      int           int      Short / Integer / Long / BigInteger
+      float         float    Float / Double
+      str           str      String
+      bin           bytes    byte[]
+      ===========   =======  =======
+
+  - Basic container types
+      ===========   =======  =======
+      MessagePack   Python   Java
+      ===========   =======  =======
+      array         list     Array
+      ===========   =======  =======
+
+  - Ray builtin types
+      - ActorHandle
+
+.. note::
+
+  * Be aware of float / double precision between Python and Java. If Java use a
+    float type to receive the input argument, the double precision Python data
+    will be reduced to float precision in Java.
+
+You can write a Python function which returns the input data:
+
+.. code-block:: python
+
+  @ray.remote
+  def py_return_input(v):
+      return v
+
+Then you can transfer the object from Java to Python, then returns from Python
+to Java:
+
+.. code-block:: java
+
+  import io.ray.api.Ray;
+  import io.ray.api.function.PyActorClass;
+  import io.ray.api.function.PyActorMethod;
+  import io.ray.api.function.PyFunction;
+  import org.testng.Assert;
+
+  public class MyRayApp {
+
+    public static void main(String[] args) {
+      Ray.init();
+
+      Object[] inputs = new Object[]{
+          true,  // Boolean
+          Byte.MAX_VALUE,  // Byte
+          Short.MAX_VALUE,  // Short
+          Integer.MAX_VALUE,  // Integer
+          Long.MAX_VALUE,  // Long
+          // BigInteger can support max value of 2^64-1, please refer to:
```

Comment:
If there is such an limitation, maybe we can undocument `BigInteger` now. As it's not very useful.

Reply:
If the `BigInteger` value out of MessagePack's range, it will raise an exception. So, I just remove this comments?